src/components/widgets/Clock.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import "../../styles/widget.css"
+import moment from 'moment'
+
+class Clock extends Component {
+
+  constructor( props ) {
+    super( props )
+    this.state = {
+      time: null,
+      amPm: null,
+      dayOfWeek: null,
+      date: null
+    }
+
+    this.setDateAndTime = this.setDateAndTime.bind(this)
+  }
+
+  setDateAndTime() {
+    let currentTime = moment().format( 'h:mm' )
+    let currentAmPm = moment().format( 'A' )
+    let currentDayOfWeek = moment().format( 'dddd' )
+    let currentDate = moment().format( 'D MMM YYYY' )
+
+    this.setState({
+      time: currentTime,
+      amPm: currentAmPm,
+      dayOfWeek: currentDayOfWeek,
+      date: currentDate
+    })
+  }
+
+  componentWillMount() {
+    this.setDateAndTime()
+  }
+
+  componentDidMount() {
+    window.setInterval( function () {
+      this.setDateAndTime()
+    }.bind( this ), 10000)
+  }
+
+  render() {
+    const { specification } = this.props
+
+    const positionStyle = { top: specification.position.x, left: specification.position.y }
+    return (
+      <article>
+        <div className="widget widget-width-1 widget-height-1" style={ positionStyle }>
+          <header>
```

Comment:
Is the header consistent among all components?  If so, we should abstract to a `Header` component